divviup/rust/src/lib.rs

```diff
@@ -147,18 +146,37 @@ pub extern "system" fn Java_org_divviup_android_Client_00024Prio3HistogramReport
     })
 }
 
+#[derive(Debug, thiserror::Error)]
+enum Error {
+    #[error("message encoding failed: {0}")]
+    Codec(#[from] prio::codec::CodecError),
+    #[error("{0}")]
+    Hpke(#[from] janus_core::hpke::Error),
+    #[error("unexpected value for jboolean")]
+    JBoolean,
+    #[error("JNI error: {0}")]
+    Jni(#[from] jni::errors::Error),
+    #[error("message decoding failed: {0}")]
+    Message(#[from] janus_messages::Error),
+    #[error("aggregator provided empty HPKE config list")]
+    MissingHpkeConfigs,
+    #[error("integer conversion error: {0}")]
+    TryFromInt(#[from] std::num::TryFromIntError),
```

Comment:
`TryFromIntError` doesn't carry any information on its own, so I think it would be worth adding back messages about which integer couldn't be converted (i.e. a parameter or the measurement) either by replacing the `TryFromIntError` with a string constant, and adding back some `.map_err()`s, or using `anyhow::Context`.